### PR TITLE
Added deprecation annotations and documentation.

### PR DESCRIPTION
**Pull request by hassler-d**
I added deprecation annotations and described how to get the same information using this library. Some are not possible yet and some are not clear, I marked them with TODOS. We should clarify those before merging.

### CODE REVIEW

File: api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java [hunk 1]
```diff
@@ -27,149 +27,205 @@
...
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	public String getSubaccountId() throws XSUserInfoException;
 
 	/**
 	 * Return the subdomain of this subaccount
 	 * 
+	 * @deprecated Can be replaced with
+	 *             {@code token.getClaimAsString(TokenClaims.XSUAA.SUBDOMAIN)} from
```
Comment (nenaraab) [inline api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: this is NOT correct / its an external attribute:
https://github.com/SAP/cloud-security-xsuaa-integration/blob/master/spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/XsuaaToken.java#L178

File: api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java [hunk 2]
```diff
@@ -27,149 +27,205 @@
...
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	public String getEmail() throws XSUserInfoException;
 
 	@Deprecated // use getHdbToken
 	public String getDBToken() throws XSUserInfoException;
 
+	// TODO 14.01.20 c5295400: can this be replaced?
```
Comment (nenaraab) [inline api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: No, have to check this with XSA 
but as it is also not supported in our Spring XsuaaToken...

File: api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java [hunk 3]
```diff
@@ -27,149 +27,205 @@
...
-	 * 
+	 *
+	 * @deprecated Can be replaced with {@code token.getAccessToken()} from the
+	 *             {@code com.sap.cloud.security.token} package.
 	 * @return authentication token
 	 */
+	@Deprecated
 	public String getAppToken();
 
+	// TODO 14.01.20 c5295400: can this be replaced?
```
Comment (nenaraab) [inline api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: No, have to check this with XSA
but as it is also not supported in our Spring XsuaaToken...

File: api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java [hunk 4]
```diff
@@ -27,149 +27,205 @@
...
 	 * Return user attributes
-	 * 
+	 *
 	 * @param attributeName
 	 *            name of attribute
 	 * @return attribute values array
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	// TODO 14.01.20 c5295400: can this be replaced?
```
Comment (nenaraab) [inline api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: Maybe we can explain it accordingly:
https://github.com/SAP/cloud-security-xsuaa-integration/blob/master/spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/XsuaaToken.java#L188

File: api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java (hunk 3 above)
Comment (nenaraab) [reply api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: Please mark with // TODO XSA?

File: api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java (hunk 2 above)
Comment (nenaraab) [inline api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: Please mark with // TODO XSA?

File: api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java [hunk 5]
```diff
@@ -27,149 +27,205 @@
...
+	 *
 	 * @return true if user attributes are available
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	// TODO 14.01.20 c5295400: can this be replaced?
 	public boolean hasAttributes() throws XSUserInfoException;
 
+
+	// TODO 14.01.20 c5295400: can this be replaced?
```
Comment (nenaraab) [inline api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: No, have to check this with XSA
but as it is also not supported in our Spring XsuaaToken...

Comment (nenaraab) [reply api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: Please mark with // TODO XSA?

File: api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java [hunk 6]
```diff
@@ -183,6 +239,7 @@
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	// TODO 14.01.20 c5295400: cannot be replaced yet (but easy to implement).
```
Comment (nenaraab) [inline api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: Maybe we can explain it according to this code snippet:
https://github.com/SAP/cloud-security-xsuaa-integration/blob/master/spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/XsuaaToken.java#L196

File: api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java [hunk 7]
```diff
@@ -192,15 +249,19 @@
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	// TODO 14.01.20 c5295400: cannot be replaced yet
```
Comment (nenaraab) [inline api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: Maybe we can explain it according to this code snippet:
https://github.com/SAP/cloud-security-xsuaa-integration/blob/master/spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/XsuaaToken.java#L201

File: api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java [hunk 8]
```diff
@@ -192,15 +249,19 @@
...
 	public String getCloneServiceInstanceId() throws XSUserInfoException;
 
 	/**
 	 * OAuth Grant Type used for this token
-	 * 
+	 *
+	 * @deprecated can be replaced with {@code token.getGrantType()} from the
```
Comment (nenaraab) [inline api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: getGrantType is of Type GrantType...

File: api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java [hunk 9]
```diff
@@ -211,20 +272,27 @@
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	// TODO 14.01.20 c5295400: can this be replaced?
```
Comment (nenaraab) [inline api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: No, have to check this with XSA
but as it is also not supported in our Spring XsuaaToken...

Comment (nenaraab) [reply api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: Please mark with // TODO XSA?

File: api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java (hunk 1 above)
Comment (hassler-d) [reply api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: Right. We have one indirection here. We could implement this though by creating something like getStringAttributeFromClaim which is possible. I correct the docs.

File: api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java (hunk 2 above)
Comment (hassler-d) [reply api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: Ok.

File: api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java (hunk 3 above)
Comment (hassler-d) [reply api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: Ok

File: api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java (hunk 8 above)
Comment (hassler-d) [reply api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: True. I add this to the docs.

File: api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java (hunk 4 above)
Comment (hassler-d) [reply api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java]: Okay, I add TODOS accordingly.

**Pull request closed without merging**